lib/practice_exercises.rb

```diff
@@ -1,13 +1,54 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n) -> It runs (n = string.lenght) times depends on the string length. 
+# Space Complexity: O(1) -> Constant variables.
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  # raise NotImplementedError, "Not implemented yet"
+  string = string.downcase
+  string = string.gsub(/[\s,:;]/ ,"")
+    i = 0
+    j = string.length - 1
+    while i < j
+      if string[i] == string[j]
+        i += 1
+        j -= 1
+      else
+        return false
+      end
+  end
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
-def longest_prefix(strings)
-  raise NotImplementedError, "Not implemented yet"
+# Time Complexity: O(n^2) Because of the nested loops. 
+# Space Complexity: O(1)
+
+def shortest_word(strings)
```

Comment:
I would say the time complexity is O(n * m) where m is the length of the shortest word and n being the number of strings.  Otherwise 👍